libraries/AP_HAL/Device.h

```diff
@@ -141,8 +146,13 @@ class AP_HAL::Device {
      */
     bool read_registers(uint8_t first_reg, uint8_t *recv, uint32_t recv_len)
     {
+        uint8_t read_reg = first_reg;
         first_reg |= _read_flag;
-        return transfer(&first_reg, 1, recv, recv_len);
+        bool result = transfer(&first_reg, 1, recv, recv_len);
```

Comment:
can you see if moving this function to the cpp file makes the flash impact lower?

Reply:
This saved quite a decent amount of flash, so I have done so for all the methods in Device.h, basically moved all the methods that decrease flash cost.